pkgs/applications/misc/octoprint/default.nix

```diff
@@ -1,91 +1,62 @@
-{ stdenv, fetchFromGitHub, python2 }:
+{ stdenv, lib, fetchFromGitHub, python2 }:
 
 let
-
-  pythonPackages = python2.pkgs.override {
-    overrides = self: super: with self; {
-      backports_ssl_match_hostname = super.backports_ssl_match_hostname.overridePythonAttrs (oldAttrs: rec {
-        version = "3.4.0.2";
-        src = oldAttrs.src.override {
-          inherit version;
-          sha256 = "07410e7fb09aab7bdaf5e618de66c3dac84e2e3d628352814dc4c37de321d6ae";
-        };
-      });
-
-      flask = super.flask.overridePythonAttrs (oldAttrs: rec {
-        version = "0.12.4";
-        src = oldAttrs.src.override {
-          inherit version;
-          sha256 = "2ea22336f6d388b4b242bc3abf8a01244a8aa3e236e7407469ef78c16ba355dd";
-        };
-      });
-
-      tornado = buildPythonPackage rec {
-        pname = "tornado";
-        version = "4.0.2";
-
-        propagatedBuildInputs = [ backports_ssl_match_hostname certifi ];
-
-        src = fetchPypi {
-          inherit pname version;
-          sha256 = "1yhvn8i05lp3b1953majg48i8pqsyj45h34aiv59hrfvxcj5234h";
-        };
-      };
-
-      flask_login = buildPythonPackage rec {
-        pname = "Flask-Login";
-        version = "0.2.2";
-
-        src = fetchPypi {
-          inherit pname version;
-          sha256 = "09ygn0r3i3jz065a5psng6bhlsqm78msnly4z6x39bs48r5ww17p";
-        };
-
-        propagatedBuildInputs = [ flask ];
-        checkInputs = [ nose ];
-
-        # No tests included
-        doCheck = false;
-      };
-
-      jinja2 = buildPythonPackage rec {
-        pname = "Jinja2";
-        version = "2.8.1";
-
-        src = fetchPypi {
-          inherit pname version;
-          sha256 = "14aqmhkc9rw5w0v311jhixdm6ym8vsm29dhyxyrjfqxljwx1yd1m";
-        };
-
-        propagatedBuildInputs = [ markupsafe ];
-
-        # No tests included
-        doCheck = false;
-      };
-
-      pylru = super.pylru.overridePythonAttrs (oldAttrs: rec {
-        version = "1.0.9";
+  mkOverride = attrname: version: sha256:
+    self: super: {
+      ${attrname} = super.${attrname}.overridePythonAttrs (oldAttrs: {
+        inherit version;
         src = oldAttrs.src.override {
-          inherit version;
-          sha256 = "71376192671f0ad1690b2a7427d39a29b1df994c8469a9b46b03ed7e28c0172c";
+          inherit version sha256;
         };
       });
     };
+
+  py = python2.override {
+    packageOverrides = lib.foldr lib.composeExtensions (self: super: { }) ([
```

Comment:
I can understand the desire to reduce code duplication, however, this won't always work because we're not always using `fetchPypi`.